### ISSUE TITLE
Template View pattern

### ISSUE THREAD

**Issue by iluwatar** [iluwatar/java-design-patterns#1320]
### Description:
The Template View design pattern is aimed at creating a consistent structure for rendering views, ensuring that the core layout and styling are maintained while allowing for specific content to be dynamic. This pattern is particularly useful for web applications where views share a common framework but differ in the specific data or content they present. The main elements of the Template View design pattern include:

- **Template Layout**: A base layout that defines the structure of the view, including headers, footers, and other common elements.
- **Dynamic Content**: Sections within the template where specific data can be injected dynamically.
- **Separation of Concerns**: Clear separation between the template (structure) and the content (data), allowing for easier maintenance and updates.

### References:
- [Martin Fowler's Template View](https://www.martinfowler.com/eaaCatalog/templateView.html)
- [P of EAA](https://books.google.fi/books?id=vqTfNFDzzdIC&pg=PA303#v=onepage&q&f=false)
- [Project Contribution Guidelines](https://github.com/iluwatar/java-design-patterns/wiki)

### Acceptance Criteria:
1. Implement a base template class that defines the overall structure of the view.
2. Create dynamic content sections that can be overridden by specific view implementations.
3. Ensure clear documentation and examples demonstrating the usage of the Template View pattern in different scenarios.

**Comment by PalAditya** [comment]
Hi @iluwatar , I would like to work on this issue if possible please
If I understand correctly, all this pattern is concerned with is creating a dynamic HTML page content based on markups, and ensuring that processing logic is delegated to the Java application, not the html page, right?
Would replicating the model shown in the link by a templating engine like Thymeleaf (with a little more added complexity and a 'bad' design alongside the correct one) be acceptable for this issue?

**Comment by iluwatar** [comment]
@PalAditya yes, I agree with your analysis. It should be sufficient to demonstrate how a templating engine is typically used to render HTML pages.

**Comment by iluwatar** [comment]
This issue is free for taking again.

**Comment by Linly1080** [comment]
Hi, I'd like to work on this issue. Could it be assigned to me if still available?

**Comment by iluwatar** [comment]
Yes @Linly1080, sorry for late reply

**Comment by Linly1080** [comment]
Sorry, I have no time work on it now.

**Comment by iluwatar** [comment]
Ok @alainalawson02

**Comment by alainalawson02** [comment]
Expected latest time to have complete by: 28/10/2022

**Comment by alainalawson02** [comment]
update: still working on this, please give us the weekend for final product. many thanks

**Comment by EthanYifanJu** [comment]
Hello @iluwatar , is this issue still available? If so could you please assign it to me? I'd happy to work on this.

**Comment by malak-elbanna** [comment]
Hi @iluwatar, I'd love to work on this issue. Can you please assign it to me?
